TRAMS-API/Controllers/ProjectsController.cs

```diff
@@ -54,10 +57,81 @@ public class ProjectsController : Controller
+        /// Academy Name</param>
+        /// <param name="status">The project status:
+        /// 1. In Progress
+        /// 2. Completed</param>
+        /// <param name="ascending">Determines if the results should be returned in ascending order. Default value is: true</param>
+        /// <param name="pageSize">The number of items to be returned per page. Must be larger than zero. Default value is: 10</param>
+        /// <param name="pageNumber">The page number to be returned. Must be larger than zero. Default value is: 1</param>
+        /// <returns></returns>
+        [HttpGet]
+        [Route("/projects")]
+        [ProducesResponseType(typeof(SearchProjectsPageModel), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status400BadRequest)]
+        public async Task<IActionResult> SearchProjects(string searchTerm, 
+                                                        ProjectStatusEnum? status,
+                                                        bool? ascending,
+                                                        uint? pageSize,
+                                                        uint? pageNumber)
+        {
+            var ascendingOption = ascending ?? true;
+            var pageSizeOption = pageSize ?? 10;
+            var pageNumberOption = pageNumber ?? 1;
+
+            if (pageSizeOption == 0)
+            {
+                return BadRequest("Page size cannot be zero");
+            }
+
+            if (pageNumberOption == 0)
+            {
+                return BadRequest("Page number cannot be 0");
+            }
+
+            Models.D365.Enums.ProjectStatusEnum projectStatus = default;
+
+            if (status.HasValue)
+            {
+                if (MappingDictionaries.ProjecStatusEnumMap.TryGetValue(status.Value, out var internalStatus))
+                {
+                    projectStatus = internalStatus;
+                }
+                else
+                {
+                    //If project status cannot be mapped to D365 Project Status, return an error
+                    return BadRequest("Project Status not recognised");
+                }
+            }
+                
+            var projSearchResult = await _projectsRepository.SearchProject(searchTerm, 
+                                                                           projectStatus,
+                                                                           ascendingOption,
+                                                                           pageSizeOption,
+                                                                           pageNumberOption);
+
+            if (!projSearchResult.IsValid)
+            {
+                return _repositoryErrorHandler.LogAndCreateResponse(projSearchResult);
+            }
+
+            var externalModels = _searchProjectsMapper.Map(projSearchResult.Result);
+
+            return Ok(externalModels);
```

Comment:
Suggestion: When implementing pagination it's common to return back paging information such as total records, current page, etc. as part of either the response header or body so that the consumer is aware of how many pages exist.  Suggest adding this information in.  Example:  https://stackoverflow.com/questions/12168624/pagination-response-payload-from-a-restful-api

Reply:
Ignore.  Can see this has been implemented.